tests/handlers/test_cas.py

```diff
@@ -201,4 +201,16 @@ def test_required_attributes(self) -> None:
 
 def _mock_request():
     """Returns a mock which will stand in as a SynapseRequest"""
-    return Mock(spec=["getClientIP", "getHeader", "_disconnected"])
+    mock = Mock(
+        spec=[
+            "finish",
+            "getClientIP",
+            "getHeader",
+            "setHeader",
+            "setResponseCode",
+            "write",
+        ]
+    )
+    # `_disconnected` musn't be another `Mock`, otherwise it will be truthy.
+    mock._disconnected = False
+    return mock
```

Comment:
`test_required_attributes`, above, results in a call to `respond_with_html_bytes`.
Previously, `_disconnected` would be a mock, which was truthy, so nothing would be written to the mock request.

Since I don't think that was intended, I've set `_disconnected` to `False` and added missing methods that now get called.

Reply:
That seems reasonable. 👍